samza-kv-rocksdb/src/main/java/org/apache/samza/storage/kv/RocksDbKeyValueReader.java

```diff
@@ -80,6 +84,33 @@ public RocksDbKeyValueReader(String storeName, String dbPath, Config config) {
     }
   }
 
+  /**
+   * Construct the <code>RocksDbKeyValueReader</code> with store's descriptor,
+   * database's path and Samza's config
+   *
+   * @param descriptor describes rocksdb
+   * @param dbPath path to the db directory
+   */
+  public RocksDbKeyValueReader(RocksDbTableDescriptor descriptor, String dbPath) {
+    keySerde = descriptor.getSerde().getKeySerde();
+    valueSerde = descriptor.getSerde().getValueSerde();
+
+    // get db options
+    ArrayList<TaskName> taskNameList = new ArrayList<TaskName>();
```

Comment:
If you do need this new constructor, then can you please consolidate the repeated logic in here and the existing constructor?